### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by JoyTaribagshaw**
- [ ] Write tests for each components.

### CODE REVIEW

File: src/tests/Navbar.test.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+test('Check navbar component rendered correctly', () => {
+  const tree = render(
+    <Provider store={store}>
+      <BrowserRouter>
+        <Navbar />
+      </BrowserRouter>
+    </Provider>,
+  );
+  expect(tree).toMatchSnapshot();
+});
```
Comment (MahmoudBakr23) [inline src/tests/Navbar.test.js:17]: - [ ] Good work writing snapshots tests so far. However, you only have 4 test cases and 1 of them is failing. Also, as per project requirements. Please make sure to test several components and pure functions related to Redux as well as integration tests of your web app. 🙏 

MINGW64__c_Users_mbakr_OneDrive_Desktop_metrics-webapp 9_9_2023 12_49_30 PM

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @JoyTaribagshaw 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Well done with the search component and functionality
- Good job with react hooks
- Great file structure
- Clean code
- Nice presentation
- Descriptive PR and README
- Correct Linters

However, there is always room for improvement.

> I'd like to remind you that your long hours of work are appreciated. However, the code review process is for improving your code, not criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

N/A

Cheers, and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/tests/Navbar.test.js (hunk 1 above)
Comment (JoyTaribagshaw) [reply src/tests/Navbar.test.js:17]: Thank you for your review, I have implemented the changes.

image

**Review by Johnadibe — APPROVED** [review]
# Status: Approved ✔️
### Hi @JoyTaribagshaw👋🏽, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

LeonardoInGatsby

## Highlights
- All tests passed✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**